### PR TITLE
coordinator::SpendTx Transaction can be null

### PR DESCRIPTION
**Pull request by danielabrozzoni**

### CODE REVIEW

File: src/message.rs [hunk 1]
```diff
@@ -173,6 +173,37 @@ pub mod coordinator {
...
+            Transaction,
+        };
+        use serde::{self, Deserialize, Deserializer, Serialize, Serializer};
+
+        pub fn serialize<S>(tx: &Option<Transaction>, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            tx.as_ref()
+                .map(|t| encode::serialize(t).to_hex())
```
Comment (darosior) [inline src/message.rs:189]: nit:
```diff
diff --git a/src/message.rs b/src/message.rs
index 9bd448a..9e0e7c9 100644
--- a/src/message.rs
+++ b/src/message.rs
@@ -176,7 +176,7 @@ pub mod coordinator {
     mod serde_tx_hex_nullable {
         use revault_tx::bitcoin::{
             consensus::encode,
-            hashes::hex::{FromHex, ToHex},
+            hashes::hex::FromHex,
             Transaction,
         };
         use serde::{self, Deserialize, Deserializer, Serialize, Serializer};
@@ -186,7 +186,7 @@ pub mod coordinator {
             S: Serializer,
         {
             tx.as_ref()
-                .map(|t| encode::serialize(t).to_hex())
+                .map(encode::serialize_hex)
                 .serialize(serializer)
         }
```

File: src/message.rs [hunk 2]
```diff
@@ -173,6 +173,37 @@ pub mod coordinator {
...
+                .serialize(serializer)
+        }
+
+        pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<Transaction>, D::Error>
+        where
+            D: Deserializer<'de>,
+        {
+            <Option<String>>::deserialize(deserializer)?
+                .map(|s| Vec::from_hex(&s).map_err(serde::de::Error::custom))
+                .transpose()?
```
Comment (darosior) [inline src/message.rs:199]: oO i didn't know about `transpose`, neat!

**Review by darosior — COMMENTED** [review]
ACK c6d278f79f390e94e64a59e73d7b76be224e9ace

**Comment by danielabrozzoni** [comment]
Updated with your nit as I liked it more :)

**Comment by darosior** [comment]
ACK 5d2c976

**Pull request merged**